mantidproject/mantid
# Add geometry information to CanSAS1D format and ISIS SANS GUI

Fixes #16705

Stephen King suggested to enhance the CanSAS1D loaders/writers to account for sample geometry information

**To test:**

Files for testing can be found here: \olympic\Babylon5\Scratch\Anton\Testing\16705_add_aperture_information_to_cansas1D
This location also contains an installer
- Open the ISIS SANS GUI (and set the Mantid path to the test files)
- Set the instrument to SANS2DTUBES
- Load the following files:

| Sample scatter | Sample trans | Sample direct |
| --- | --- | --- |
| SANS2D00034481 | SANS2D00034502 | SANS2D00034461 |
- Run the 1D reduction
- Make sure that the workspace 34481rear_1D_1.75_16.5 exists
- Press Save Result, but make sure that only CanSAS is selected as the format
  - Confirm that an xml file was saved in your output directory
  - Inspect the xml file. Confirm that it contains entires under SASsample->thickness
  - Inspect the xml file. Confirm that it contains entires under SASinstrument->SAScollimation->name
  - Inspect the xml file. Confirm that it contains entires under SASinstrument->SAScollimation->X
  - Inspect the xml file. Confirm that it contains entires under SASinstrument->SAScollimation->Y
- Open the LoadCanSAS1D dialog window
- Select the xml file as the input file
- Select the name "test" as the output
- Run the LoadCanSAS1D algorithm
- Use the script below to confirm that the values on the workspace are correct. Note that the saved out values can be found on the Run Settings Tab under the SampleDetails section

``` python
def convertFlagToName(flag):
    if flag == 1:
        return "Cylinder"
    elif flag == 2:
        return "Flat plate"
    elif flag ==3:
        return "Disc"
    else:
        return "Something is wrong"

ws = mtd["test"]
sample = ws.sample()
print "The geometry flag indicates: {}".format(convertFlagToName(sample.getGeometryFlag()))
print "The thickness is {}".format(sample.getThickness())
print "The width is {}".format(sample.getWidth())
print "The height is {}".format(sample.getHeight())
```
- Confirm that the values output by the script are identical to the ones on the Geometry tab.
- Delete the XML file; Delete the "test" workspace
- Press "Save Others"; select only 34481rear_1D_1.75_16.5 and the CanSAS format; press Save
- For verifaction follow the same steps as above

Relese notes are [here]

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
